app/controllers/character_weapons_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CharacterWeaponsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_character_weapon, except: [:new, :create]
+  before_action :verify_character_weapon, except: [:new, :create]
+
+  def destroy
+    @character_weapon.destroy
+    redirect_to @character_weapon.character, flash: { notice: "Weapon #{@character_weapon.weapon.name} has been destroyed" }
+  end
+
+  def show
+    @weapon = @character_weapon.weapon
+  end
+
+  def new
+    @weapons = Weapon.order_by_rarity.order_by_price
+    @character = Character.find(params[:character_id])
+  end
+
+  def create
+    character = Character.find(params[:character_id])
+    weapon = Weapon.find(params[:weapon_id])
+    @character_weapon = CharacterWeapon.new(character: character, weapon: weapon, damage: weapon.default_damage, crit: weapon.default_crit)
+
+    if @character_weapon.save
+      redirect_to @character_weapon.character, notice: "Weapon #{@character_weapon.weapon.name} was successfully created."
```

Comment:
Why don't you use the local variables `character` & `weapon` instead of fetching them again?